Release: npm@6.5.0

zkat opened the pull request:
[rendered changelog](https://github.com/npm/cli/blob/release-next/CHANGELOG.md)

bakkot committed:
docs: mention --otp flag when prompting for OTP (#60)

PR-URL: https://github.com/npm/cli/pull/60
Credit: @bakkot
Reviewed-By: @zkat

maartenba committed:
audit: set lowercase headers for npm audit requests (#62)

Fixes: https://npm.community/t/npm-audit-making-non-rfc-compliant-requests-to-server-resulting-in-400-bad-request-pr-with-fix/1742
PR-URL: https://github.com/npm/cli/pull/62
Credit: @maartenba
Reviewd-By: @zkat

zkat committed:
ci: backronym ci to clean-install (#57)

Credit: @zkat
Reviewed-By: @jefflembeck

zkat committed:
version: fix sign-git-commit options (#58)

Oops: 7984206e2f41b8d8361229cde88d68f0c96ed0b8 did the
wrong thing with the git params, and it breaks anyone
trying to sign their git commits with npm version.

Fixes: https://npm.community/t/1661
Credit: @zkat

zkat committed:
figgy-pudding@3.5.1

Credit: @zkat

zkat committed:
ssri@6.0.1

Fixes: https://npm.community/t/crash-invalid-config-key-requested-error/1764
Credit: @zkat

iarna committed:
bluebird@3.5.2

iarna committed:
ci-info@1.5.1

iarna committed:
config-chain@1.1.12

iarna committed:
glob@7.1.3

iarna committed:
opener@1.5.1

Fix browser opening under Windows Subsystem for Linux (WSL).

Credit: @thijsputman

iarna committed:
semver@5.5.1

Commit:
docs: clarify default branch with git url (#64)

With some tests I note npm use default branch of repository instead of `master`.

PR-URL: https://github.com/npm/cli/pull/64
Credit: @zckrs
Reviewed-By: @zkat

kemitchell committed:
test: Automatically audit dependency licenses (#70)

PR-URL: https://github.com/npm/cli/pull/70
Credit: @kemitchell
Reviewed-By: @zkat

RobertKielty committed:
docs: bash_completion.d dir found in /etc not /usr/local (#72)

I found this page useful for setting up npm command completion. Thank you npm 
documentation team.

This suggested changed is based on my system which runs Ubuntu where
the completion directory path for bash is found in 
/etc/bash_completion.d and not /usr/local/etc/bash_completion.d

I dug around docs for this in The Bash Manual 
https://www.gnu.org/software/bash/manual/bashref.html#Programmable-Completion 
and on some an old (sadly retired) Debian Admin Site
https://debian-administration.org/article/317/An_introduction_to_bash_completion_part_2

which seemed supportive of the dir being in /etc but I do not have access 
to an OS X machine where I guess bash_completion.d could be in /usr/local/etc 
so I left it in.

PR-URL: https://github.com/npm/cli/pull/72
Credit: @RobertKielty
Reviewed-By: @zkat

scotttrinh committed:
docs: Update OTP documentation for dist-tag add (#74)

See discussion here: https://npm.community/t/npm-dist-tag-add-with-2fa-enabled-fails-for-non-latest-tag-with-500/2432

TL;DR:

> To be clear, you can specify an existing tag and it’ll change it. If you have 2FA enabled, you do need to specify `--otp` and if you don’t, you should get a `401` as above.

Ref: https://npm.community/t/npm-dist-tag-add-with-2fa-enabled-fails-for-non-latest-tag-with-500/2432
PR-URL: https://github.com/npm/cli/pull/74
Credit: @scotttrinh
Reviewed-By: @iarna
Reviewed-By: @zkat

larsgw committed:
edit: scoped packages (#75)

* edit: fix handling of scoped packages

* edit: fix usage info

* docs: fix docs for the npm-edit command

PR-URL: https://github.com/npm/cli/pull/75
Credit: @larsgw
Reviewed-By: @iarna

hugovk committed:
docs: Correct spelling (#78)

PR-URL: https://github.com/npm/cli/pull/78
Credit: @hugovk
Reviewed-By: @zkat

alopezsanchez committed:
ci: change `console.error` for `console.log` (#79)

Fixes: https://npm.community/t/npm-ci-logs-success-to-stderr/2303
PR-URL: https://github.com/npm/cli/pull/79
Credit: @alopezsanchez
Reviewed-By: @zkat

macdja38 committed:
publish: Close the file descriptor if exiting upload via an error. (#80)

This fixes https://npm.community/t/unhelpful-error-message-when-publishing-without-logging-in-error-eperm-operation-not-permitted-unlink/1377/3 and the other dozen or so issues that that link references, and possibly many more involving poor error messages from errors thrown by the upload function. 

@zkat you mentioned you could take a look at any fixes / answer any questions, if you could look this over and let me know if this is a good / valid approach that would be fantastic, thanks! (it wasn't a race condition, luckily :P).

it may also be helpful to add something like 
```
    if (!auth.token || !(auth.username && auth.password)) {
      log.warn('publish', 'not logged in')
    }
```
just before we even open the first file descriptor to make sure that even if the error message is completely wrong something in the log will give users a clue what may be going on. I took the method of looking for login creds from the logout method, I'm not sure that's valid or if alternatives to npm exist that don't require credentials but users could still publish to.

Triage of the issue:

1. The upload function throws an error
2. As that error bubbles through [cacache](https://www.npmjs.com/package/cacache#with-tmp) it tries to delete the tmpdir as it should
3. It can't delete the temp dir as the upload function's readFileStream to the tar it was trying to upload is still open.
4. [cacache](https://www.npmjs.com/package/cacache#with-tmp) throws an error about it's inability to remove the dir, which suppresses the upload function's error.

Fixes: https://npm.community/t/unhelpful-error-message-when-publishing-without-logging-in-error-eperm-operation-not-permitted-unlink/1377/3
PR-URL: https://github.com/npm/cli/pull/80
Credit: @macdja38
Reviewed-By: @zkat

Commit:
cli,outdated: Adds 'Homepage' to outdated --long output. (#81)

* feat(cli, outdated): Adds 'Homepage' to outdated --long output.

- `package.json`'s `homepage` property is displayed when using the `--long` option for `npm outdated`

* test: npm outdated --long

- Adds `homepage` to `--parseable` output.
- Updates `npm outdated --long` test to include `homepage` in expected output.
- Adds `homepage` to `npm-outdated` documentation.

* fix: javascript standard style updates

PR-URL: https://github.com/npm/cli/pull/81
Credit: @jbottigliero
Reviewed-By: @zkat

seishun committed:
docs: Note that "prepare" runs when installing git dependencies (#82)

PR-URL: https://github.com/npm/cli/pull/82
Credit: @seishun
Reviewed-By: @zkat

kjin committed:
docs: Specify that --dry-run isn't available in older versions of npm publish (#83)

Not sure when `--dry-run` was introduced, I assume it was introduced in npm 6 (correct me if I'm wrong). Anyway, hopefully this will prevent future developers from making the mistake of using it in npm 5 (like I did).

PR-URL: https://github.com/npm/cli/pull/83
Credit: @kjin
Reviewed-By: @zkat

kitayoshi committed:
docs: fix tag in inline code (#96)

PR-URL: https://github.com/npm/cli/pull/96
Credit: @midare
Reviewed-By: @zkat

neverett committed:
docs: Add semver link and note on empty string format to `deprecate` doc (#68)

PR-URL: https://github.com/npm/cli/pull/68
Credit: @neverett
Reviewed-By: @zkat

zkat committed:
bluebird@3.5.3

Credit: @petkaantonov

zkat committed:
graceful-fs@4.1.15

Credit: @isaacs

zkat committed:
tar@4.4.8

Credit: @isaacs

zkat committed:
npm-packlist@1.1.12

Credit: @isaacs

zkat committed:
ci-info@1.6.0

Credit: @watson

zkat committed:
doc: fix semver docs

zkat committed:
license: whitelist config-chain again

zkat committed:
deps: gitignore licensee and deps

zkat committed:
doc: update changelog for npm@6.5.0

zkat committed:
update AUTHORS

zkat committed:
6.5.0-next.0

booc0mtaco commented:
Hey there,

Was digging around, and spotted some busted formatting on a page. 

https://docs.npmjs.com/misc/scripts

Breaking chunk is [here](https://github.com/npm/cli/blame/latest/doc/misc/npm-scripts.md#L56), i think. perhaps `<pkg>` needs to be brought up to the previous line (or there's a bug in the markdown processor).

aeschright commented:
> Hey there,
> 
> Was digging around, and spotted some busted formatting on a page.
> 
> https://docs.npmjs.com/misc/scripts
> 
> Breaking chunk is [here](https://github.com/npm/cli/blame/latest/doc/misc/npm-scripts.md#L56), i think. perhaps `<pkg>` needs to be brought up to the previous line (or there's a bug in the markdown processor).

Fixed in #96 for this release.

iarna committed:
chore: fix name of fake-registry docs

iarna committed:
unsupported: Add v12 (nightlies) to supported list

iarna committed:
travis: Add node@11 to test matrix

iarna committed:
travis: only run license check once per full test run (like standard)

iarna committed:
travis: 10 is LTS now

Commit:
6.5.0

dmabupt committed:
config: support default user on IBM i (#65)

setuid() can not accept a 'nobody' parameter on IBM i.
The default user (QSECOFR) on IBM i has user id 0.

PR-URL: https://github.com/npm/cli/pull/65
Credit: @dmabupt
Reviewed-By: @zkat

GeorgeTaveras1231 committed:
docs: Fix typo in npm-token documentation (#71)

PR-URL: https://github.com/npm/cli/pull/71
Credit: @GeorgeTaveras1231
Reviewed-By: @zkat

larsgw committed:
pacote: pass git opts to pacote (#98)

Pass git binary path config to pacote.

Ref: zkat/pacote#164
Fixes: https://npm.community/t/3278
PR-URL: https://github.com/npm/cli/pull/98
Credit: @larsgw
Reviewed-By: @zkat

isaacs committed:
dist-tag: make 'ls' the default action (#106)

I keep typing `npm dist-tags` expecting it to print out a list of
dist-tags and instead it yells at me and that feels very un-npm-y.

PR-URL: https://github.com/npm/cli/pull/106
Credit: @isaacs
Reviewed-By: @zkat

BeniCheni committed:
utils: check npm.config existence in error-handler.js (#108)

Fixes: https://npm.community/t/npx-envinfo-preset-jest-fails-on-windows-with-a-stack-trace/2713
PR-URL: https://github.com/npm/cli/pull/108
Credit: @BeniCheni
Reviewed-By: @zkat
